src/App.jsx

```diff
@@ -14,45 +16,137 @@ export const goodsFromServer = [
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+
+        case SORT_FIELD_BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return makeReverse ? preparedGoods.reverse() : preparedGoods;
+}
+
+export const App = () => {
+  const [visibleGoods, setVisibleGoods] = useState(goodsFromServer);
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [isReset, setIsReset] = useState(false);
+
+  const resetClick = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    setIsReset(false);
+    setIsReversed(false);
+  };
+
+  const sortAlphabeticallyClick = () => {
+    if (isReversed) {
+      setVisibleGoods(sortGoods(visibleGoods, SORT_FIELD_ALPHABETICALLY, true));
+    } else {
+      setVisibleGoods(sortGoods(visibleGoods, SORT_FIELD_ALPHABETICALLY));
+    }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    setSortField(SORT_FIELD_ALPHABETICALLY);
+    setIsReset(true);
+  };
+
+  const sortByLengthClick = () => {
+    if (isReversed) {
+      setVisibleGoods(sortGoods(goodsFromServer, SORT_FIELD_BY_LENGTH, true));
+    } else {
+      setVisibleGoods(sortGoods(visibleGoods, SORT_FIELD_BY_LENGTH));
+    }
+
+    setSortField(SORT_FIELD_BY_LENGTH);
+    setIsReset(true);
+  };
+
+  const reverseClick = () => {
+    setVisibleGoods(sortGoods(visibleGoods, '', true));
+    setIsReversed(!isReversed);
+    setIsReset(true);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-info',
+            {
+              [isLight]: sortField !== SORT_FIELD_ALPHABETICALLY,
+            },
+          )}
+          onClick={sortAlphabeticallyClick}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-success',
+            {
+              [isLight]: sortField !== SORT_FIELD_BY_LENGTH,
+            },
+          )}
+          onClick={sortByLengthClick}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            'button',
+            'is-warning',
+            {
+              [isLight]: !isReversed || !isReset,
+            },
+          )}
+          onClick={reverseClick}
+        >
+          Reverse
+        </button>
+
+        {isReset
+        && JSON.stringify(visibleGoods) !== JSON.stringify(goodsFromServer)
```

Comment:
please don't call a function from the interface because it will be executed every time when the state change, so make sense to move it to the variable